config/src/lib.rs

```diff
@@ -56,106 +80,199 @@ pub struct ConfigOption {
+    pub fn build<T: Configurable>(
+        commandline: Vec<OsString>,
+        config_file_env: &str,
+    ) -> Result<T, Error> {
+        // Parse commandline according to config definition
+        // -> ParsedValues
+        let description = T::get_config_description();
+
+        //
+        let matcher = ConfigBuilder::create_matcher(&description);
+
+        // Overwrite command line arguments with final arguments to parse
+        // if a config file was specified, all options from that file will be
+        // prepended to the command line arguments
+        let commandline =
+            ConfigBuilder::maybe_combine_arguments(matcher.clone(), commandline, config_file_env);
+
+        // Parse command line
+        let test = commandline.clone().expect("ie");
+        println!("Commandline: {:?}", test);
+        let matcher = matcher.get_matches_from(commandline.expect("trnsi"));
+
+        // Convert results from command line parsing into a HashMap<ConfigOption, Vec<String>>
+        // this is then passed to the actual implementation of the configuration for processing
+        let mut result: HashMap<ConfigOption, Option<Vec<String>>> = HashMap::new();
+        for config_option in description.options.clone() {
+            println!(
+                "{}: is_present(): {} - occurrences_of(): {}",
+                config_option.name,
+                matcher.is_present(config_option.name),
+                matcher.occurrences_of(config_option.name)
+            );
+            if matcher.occurrences_of(config_option.name) != 0 {
+                let found_values = matcher
+                    .values_of(config_option.name)
+                    .expect("error")
+                    .collect::<Vec<&str>>();
+
+                // Convert to Vec of owned Strings, as we will want to keep these values around for
+                // the lifetime of our application
+                let found_values: Vec<String> =
+                    found_values.into_iter().map(String::from).collect();
+                result.insert(config_option, Some(found_values));
+            } else {
+                // Option was not specified on the command line, so instead of adding an empty
+                // Vec we add _None_ in order to be able to distinguish between  these cases
+                // scenarios later on
+                result.insert(config_option, None);
+            }
+        }
+        // Return an actual object of the configuration that is populated with appropriate values
+        Ok(T::parse_values(result))
+    }
+
+    // Create a clap matcher based on the ConfigOptions that were defined in the config object
+    fn create_matcher(config: &Configuration) -> App {
+        let mut matches = App::new(config.name)
+            .version(config.version)
+            .about(config.about);
+
+        for option in config.options.iter() {
+            let mut new_arg = Arg::with_name(option.name)
```

Comment:
The compiler issues a warning for this line. I wonder why we need a `mut`able variable here.

Reply:
Mental error on my part, I added the _mut_, when I refactored the code a few lines further down to conditionally allow multiple arguments or not. 
I originally had the "if option.list {" part as mutating the new_arg variable and at that time needed the mut.

Removed it.